apps/openmw/mwgui/journalbooks.cpp

```diff
@@ -217,25 +219,37 @@ book JournalBooks::createQuestBook (const std::string& questName)
 }
 
 book JournalBooks::createTopicIndexBook ()
+{
+    ToUTF8::FromType encoding = MWBase::Environment::get().getWindowManager()->getEncoding();
+    bool isRussian = (encoding == ToUTF8::WINDOWS_1251);
+
+    BookTypesetter::Ptr typesetter = isRussian ? createCyrillicJournalIndex() : createLatinJournalIndex();
+
+    return typesetter->complete ();
+}
+
+BookTypesetter::Ptr JournalBooks::createLatinJournalIndex ()
 {
     BookTypesetter::Ptr typesetter = BookTypesetter::create (92, 250);
 
     typesetter->setSectionAlignment (BookTypesetter::AlignCenter);
 
-    BookTypesetter::Style* body   = typesetter->createStyle ("", MyGUI::Colour::Black);
-
+    BookTypesetter::Style* body = typesetter->createStyle ("", MyGUI::Colour::Black);
     for (int i = 0; i < 26; ++i)
     {
         char ch = 'A' + i;
-
         char buffer [32];
-
         sprintf (buffer, "( %c )", ch);
 
+        char buffer2 [32];
+        sprintf(buffer2, "%c", ch);
+        Utf8Stream stream (buffer2);
+        uint32_t first = stream.peek();
```

Comment:
Wouldn't it be enough to just cast `ch` to uint32? I would expect that ASCII values are the same in Unicode32.